cmd/xl-storage.go

```diff
@@ -1202,7 +1208,12 @@ func (s *xlStorage) cleanupTrashImmediateCallers(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case entry := <-s.immediatePurge:
-			removeAll(entry)
+			// Add deadlines such that immediate purge is not
+			// perpetually hung here.
+			w := xioutil.NewDeadlineWorker(globalDriveConfig.GetMaxTimeout())
+			w.Run(func() error {
+				return removeAll(entry)
```

Comment:
For a more timely response, don't we also want to propagate (and react to) a context?

Reply:
hmmm.. I guess we'd need to fork the stdlib :/

Reply:
yeah no way to do it